src/main/java/carpetextra/mixins/FlowerPotBlockMixin.java

```diff
@@ -40,7 +36,7 @@ public FlowerPotBlockMixin(Settings settings)
         ),
         locals = LocalCapture.CAPTURE_FAILHARD
     )
-    private void onActivate(BlockState state, World world, BlockPos pos, PlayerEntity player, Hand hand, BlockHitResult hit, CallbackInfoReturnable<ActionResult> cir, ItemStack itemStack, Item i, BlockState blockState) {
+    private void onActivate(BlockState blockState, World world, BlockPos pos, PlayerEntity player, BlockHitResult hit, CallbackInfoReturnable<ActionResult> cir) {
```

Comment:
half broken: this works only when activated from a dispenser (with dispenserspotflowers ON), not when planting a rose by hand in the pot.

Reply:
https://github.com/voidpointer0x00/carpet-extra/pull/2